kie-server-parent/kie-server-services/kie-server-services-optaplanner/src/main/java/org/kie/server/services/optaplanner/OptaplannerKieServerExtension.java

```diff
@@ -86,18 +83,33 @@ public void init(KieServerImpl kieServer, KieServerRegistry registry) {
         // The following thread pool will have a max thread count equal to the number of cores on the machine minus 2,
         // leaving a few cores unoccupied to handle REST/JMS requests and run the OS.
         // If new jobs are submitted and all threads are busy, the default reject policy will kick in.
-        int availableProcessorCount = Runtime.getRuntime().availableProcessors();
-        int resolvedActiveThreadCount = Math.max(1, availableProcessorCount - 2);
-        int queueSize = Integer.parseInt(System.getProperty(
-                KieServerConstants.KIE_OPTAPLANNER_THREAD_POOL_QUEUE_SIZE, String.valueOf(resolvedActiveThreadCount)));
-        logger.info("Creating a ThreadPoolExecutor with corePoolSize = " + resolvedActiveThreadCount + ","
-                + " maximumPoolSize = " + resolvedActiveThreadCount + ", queueSize = " + queueSize);
+        //int availableProcessorCount = Runtime.getRuntime().availableProcessors();
+        //int resolvedActiveThreadCount = Math.max(1, availableProcessorCount - 2);
+
+        String corePoolSize = System.getProperty("corePoolSize").isEmpty() ? System.getProperty("corePoolSize") : "2";
+        String maxPoolSize = System.getProperty("maxPoolSize").isEmpty() ? System.getProperty("maxPoolSize") : "10";
+        String keepAliveTimeSeconds = System.getProperty("keepAliveTimeSeconds").isEmpty() ? System.getProperty("keepAliveTimeSeconds") : "10";
+
         this.threadPool = new ThreadPoolExecutor(
-                resolvedActiveThreadCount,
-                resolvedActiveThreadCount,
-                10, // thread keep alive time
+                Integer.parseInt(corePoolSize),
+                Integer.parseInt(maxPoolSize),
+                Integer.parseInt(keepAliveTimeSeconds), // thread keep alive time
                 TimeUnit.SECONDS,
-                new ArrayBlockingQueue<>(queueSize)); // queue with a size
+                new LinkedBlockingQueue<>()); // queue with a size
+
+        StringBuilder s = new StringBuilder();
+        s.append("poolSize = ")
+                .append(this.threadPool.getPoolSize())
+                .append(", corePoolSize = ")
+                .append(this.threadPool.getCorePoolSize())
+                .append(", queueSize = ")
+                .append(this.threadPool.getQueue().size())
+                .append(", queueRemainingCapacity = ")
+                .append(this.threadPool.getQueue().remainingCapacity())
+                .append(", maximumPoolSize = ")
+                .append(this.threadPool.getMaximumPoolSize());
+
+        logger.info(s.toString());
```

Comment:
This is too verbose to output on info logging. Let's make this debug logging.

Reply:
No need to use StringBuilder here, it's equally fast as just String concatenation because there is no loop. (faster even, see shipilev's blog).
Furthermore, better to use logger.debug("OptaPlanner extension created with corePoolSize ({}), queueSize ({}), ...", corePoolSize, queueSize, ...)